app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt

```diff
@@ -224,10 +224,15 @@ class AutoCompleteApi @Inject constructor(
                 AutoCompleteSearchSuggestion(phrase = it.phrase, isUrl = (it.isNav ?: UriString.isWebUrl(it.phrase)))
             }
             .toList()
-            .onErrorReturn {
-                if (it is InterruptedIOException) throw it else emptyList<AutoCompleteSearchSuggestion>()
-            }
             .toObservable()
+            .onErrorResumeNext { throwable: Throwable ->
+                if (throwable is InterruptedIOException) {
+                    // When interrupted, return an empty observable to avoid showing the default state
```

Comment:
Can we explain a bit more what "interrupted" means for more clarity? i.e. quickly deleting all text which would cancel the subscription

Reply:
Updated:
```
// If the query text is deleted quickly, the request may be cancelled, resulting in an InterruptedIOException.
// Return an empty observable to avoid showing the default state.
```